### ISSUE TITLE
New Layout Preview

### ISSUE THREAD

**Issue by ElementalCrisis** [japanesemediamanager/japanesemediamanager.github.io#12]
Hello Everyone,

While working on the current site I created a to do list of things I wanted to add to enhance the site and provide additional functionality. However I decided that instead of trying to add to the current layout, the best option would be to find a better layout. After finding a couple different ones and talking with @jmediamanager I found one that's setup very similar to what I wanted to originally do with the original layout. 

I've put together a test site where I've been moving all the information over and wanted to get some feedback and address any bugs before the new layout is launched. 

Here's a list of the new features that will launch with the layout.

**Responsive Site**
No more fixed width for all resolutions. Using the data provided by W3 Schools I've increased the default width allowing more content to be shown on screen. Those with a resolution smaller than the new default with still see the same site. Additionally mobile browsing support has been added, the site will automatically resize itself to your phones screen. 

**Link Navigation**
No more clicking the back button to get to another page! A sticky menu has been added that will follow you as you scroll. In addition I've separated a lot of the pages into their own page for clearer navigation.

**Displayed Image Enchantments**
Currently you have to create a thumbnail for your image or the image breaks the page. You no longer have to do this as the image is automatically scaled to the pages width. Lightbox has also been added allowing images to be displayed on top of the page instead of in a new window.

Also related to images, I've gone through and compressed a lot of them allowing for faster loading with minimal quality loss. I'm also testing out a new naming method replacing underscores with dashes and making the filenames reflect the image so let me know what you think.

I still have a few things to add, most likely not until after the new layout launches. 

Here's the link to check out the new layout, not everything has been transferred over so there might be some links that take you nowhere. I'll be pushing an update later tonight that has more content transferred over.

http://elementalcrisis.github.io/

**Comment by xsystematic** [comment]
Looks good, will take a more in depth look in the next few days

**Comment by ElementalCrisis** [comment]
Thank you, I appreciate it. Question for you and anyone else reading. Which style for the section titles looks better?

http://elementalcrisis.github.io/support.html or http://elementalcrisis.github.io/contribute.html

Does the big section titles look better or the small bold section titles?

**Comment by AreFinstad** [comment]
the support page looks much better IMO, it also takes up less space.
text does not need to be that large in almost incase IMO, top of the page to announce the subject it fine tho IMO.

also inputting on colors, if you can change styles and such i would really love if we had a midnight mode (dark mode) to choose from as it is now its too bright for brighter screens and so on, i should not need to use f.lux or the stylish chrome app for it to be readable without hurting my eyes...... esp in the middle of night XD
just my thoughts tho.


in other input, on the main screen we should have picture carousels that have more than just one picture to show the apps more in depth, meaning someone has to take some kickass new shots to use.

**Comment by ElementalCrisis** [comment]
Thanks for the feedback.

> the support page looks much better IMO, it also takes up less space.
> text does not need to be that large in almost incase IMO, top of the page to announce the subject it fine tho IMO.

The MySQL setup page (http://elementalcrisis.github.io/setup/mysql.html) is how I think I'll set pages up. It's very similar to the support layout.

> also inputting on colors, if you can change styles and such i would really love if we had a midnight mode (dark mode) to choose from

Additional themes are on my list, just not till after the new site is up and running. But I'll see if I can get a dark theme before then, feel free to link to some examples. 

> in other input, on the main screen we should have picture carousels that have more than just one picture to show the apps more in depth, meaning someone has to take some kickass new shots to use.

It's on my list for a future update, I don't use My Anime 3 or Plex Plugin so I'll need people to submit those images but I can do the rest.

**Comment by AreFinstad** [comment]
can be easy enough to get some for you there. just shout when you want them and what you would like them to be of.

**Comment by ElementalCrisis** [comment]
Feel free to design the dark theme however you think it would look best. Here's my version of a dark theme. 
![jmms - dark theme](https://cloud.githubusercontent.com/assets/9443295/9957879/cbc498fc-5db8-11e5-895e-3514c588680a.png)

**Comment by ElementalCrisis** [comment]
Here's a good site to get color schemes.
http://paletton.com

**Comment by AreFinstad** [comment]
well if you want to still have the blue scheme then for a dark/midnight theme it would prob use something around #1F2025 for the background and #D6D6D6 for the text while keeping the bright blue vanilla link text. but thats just me tho XD

which to be fair is pretty close to what you have there TBH XD
i do think the sideline bar should be a bit darker tho.... well maybe not, i just feels its a bit out of place. 

![](http://i.imgur.com/FAL7XnK.png)

**Comment by ElementalCrisis** [comment]
You really want a blank theme, lol. 

I ended up adding in the theme selector, for now it's only visible on this page. 
http://elementalcrisis.github.io/general/faq.html

**Comment by AreFinstad** [comment]
haahah you should see my github with stylish XD 

![](http://i.imgur.com/8ExBi9l.png)


yeah, i like it dark as my day rhythm is messed up most of the time and in most active after midnight.
simple dark textures also works great too IMO as it doesnt make it as flat as just a color code.

**Comment by xsystematic** [comment]
I like the dark theme @ElementalCrisis, much easier on my eyes at night.

**Comment by ElementalCrisis** [comment]
Changes:
- Dark Theme is now globally supported.
- Fixed some CSS/JS link errors. 
- Finished the General section for the Documentation.
- Added To-Do and Guide page. 

I'm working on creating a guide people can use to help them when adding to the website and a to-do page that list's what still needs to be done. I've added a few things to the guide which you can access under contribute or by this link, http://elementalcrisis.github.io/guide.html, let me know if it's easy to understand and would be helpful if you were adding to the website. 

I still haven't messed around with the darker colors yet @AreFinstad, I'll work on that next.

**Comment by ElementalCrisis** [comment]
So I wanna try something. On the Light Theme I changed the width to 100%, basically meaning the page scales to the windows width. How does this look for everyone?

**Comment by AreFinstad** [comment]
works fine on a 9:16 display, most sites does not work as smooth on it.

not that the Success alert and error alerts are not centered like the other two tho, something they should be IMO. 

![](http://i.imgur.com/hngCtDw.png)

**Comment by ElementalCrisis** [comment]
I set the alerts up like that to see how it scales. Glad the page looks fine for you, I tested multiple resolutions and it looked fine as well. I'll make the changes to the Dark Theme later.

**Comment by ElementalCrisis** [comment]
@AreFinstad, what program are you using to take a full width screenshot?

**Comment by AreFinstad** [comment]
app actually :D 
this one: https://chrome.google.com/webstore/detail/webpage-screenshot/bdfnieppndfdhcgbmhfdlgdjegclkomk

only thing is that the sharing page for it is kinda bad so i always just download it from that site to upload with ShareX to imgur or whatever :D (BTW ShareX is the best sharing tool i have ever seen and should be used before anything else IMO)

**Comment by ElementalCrisis** [comment]
Changes:
- Anime Buddy pages complete.
- Mobile menu added along with mobile only links. 

I'm nearly done. If anyone wants to help write some of the missing JMM pages but doesn't know how to code just type it out and I can input it and fix it up.

**Comment by AreFinstad** [comment]
hmmm just noticed on the AnimeBuddy description that it says it can stream..... last time i checked it cant stream but only download the original video file so it can be played locally.
cant really confirm this anymore as it fails to log in on all points for me ATM.

**Comment by ElementalCrisis** [comment]
I'll double check when I get home, I think there's a button labeled stream though. Can you list the images with compression issues? Most of them are saved as JPG, I'll probably go through and re-create a lot of the images as PNG.

**Comment by ElementalCrisis** [comment]
Yep, it's called streaming. 
![ab - options](https://cloud.githubusercontent.com/assets/9443295/10034902/daa888da-6149-11e5-86cb-2e3ce0fb6369.png)
You're right in the fact that it just downloads the file and puts it in the temp directory though. I'll make a note of this.

**Comment by jmediamanager** [comment]
Ah yeah, I need to remove all references to streaming.

I actually did some worl along those lines, but couldn't get it to work reliably.
Still would prefer to have it one day.

**Comment by ElementalCrisis** [comment]
Alright I added a sticky menu to the site. I took the main links off the scroll menu and put them on the sticky menu, the scroll menu will now be just used for page links. 

Thoughts on this?

Also this is going to be the last new feature I add, after this I'm going to get the rest of the content transferred over to finish with the new site.

**Comment by AreFinstad** [comment]
love the new sidemenu, works great. if its like that across all pages then great :D

**Comment by ElementalCrisis** [comment]
it's site-wide on my local version, will be up in the next update. 

Can someone test something for me? Go to http://elementalcrisis.github.io/ and open the images and let me know how they look. I want to replace older images with higher resolution ones and since the index already has them it's a good place to get an idea on how things will look.

**Comment by AreFinstad** [comment]
it looks good so far, i just think its a bit slow tho, the fadetime is a bit long for my taste, unsure if that is something you can change tho.

other than that it looks pretty good so far.

also you should have the image window constrained to the browser size as large images might become an issue for smaller screens and so on, and ofc if we take huge shots for some reason....

**Comment by ElementalCrisis** [comment]
Hmm, the fadetime is pretty fast for me, can you make a video so I can see? Also, the way it's setup it wont stretch the user's browser even on large images.

**Comment by xsystematic** [comment]
Just a quick note for the mysql setup page. I Just rebuilt my htpc and used the guide on site to set up mysql but failed due to a bug in the mysql installer. I used the mysql-installer-web-community-5.6.26.0.msi

Bug and workaround can be found here
https://bugs.mysql.com/bug.php?id=77965

The problem with this bug is that when you try to install mysql with a windows account that has a space in the name (such as Test Test)  it asks for an update that ends up failing. I didn't try installing without the update though so cant confirm if it works that way. 

Thought I should let you know as anyone new to mysql who follows the guide on site may run into problems.

**Comment by AreFinstad** [comment]
plz make new issue about your problem at https://github.com/japanesemediamanager/jmmserver/issues

so we can track it and such.

**Comment by xsystematic** [comment]
I posted it here as the problem is with the guide and the mysql installer not jmm server, just thought that you could add a note to the guide itself

**Comment by AreFinstad** [comment]
sounds fair enough.

**Comment by ElementalCrisis** [comment]
@xsystematic Thanks for the info, I'll make the changes and link to that thread.

**Comment by ElementalCrisis** [comment]
Alright, I just updated the test site and with this update I've finished moving everything over. If I could ask @AreFinstad and @xsystematic to go over it and see if I missed anything or if there are any problems I would really appreciate it. Once down I'll submit a commit to the live site for @jmediamanager to look over and approve. 

Additionally, anyone looking to help can head over to http://elementalcrisis.github.io/todo.html to see a list of things that still need to be done.

**Comment by ElementalCrisis** [comment]
Also, I meant to remove it but my test link with the chat option for Discord is on the index page. As @jmediamanager hasn't approved this yet it wont be on the live site but feel free to join to talk with @AreFinstad and I.

**Comment by xsystematic** [comment]
On the latest downloads page the beta download link fails as the file has been deleted. 
On the installing JMM Desktop and JMM Server pages, at the bottom of the page it says use the menu on the right. It should say on the left.

Also I feel that any links to different sites should open in a new tab. Currently it's a mix of opening in the same tab or opening in a new tab. An example would be on the home page "What is JMM" on the "How does JMM work? section, the links to AniDB and TheTvDB both open in the same tab but the  TMDb link opens in a new tab.
I'm using the latest Chrome browser, thought I would mention my browser as it may just be a Chrome issue.

Good to see the site is progressing well and I will be looking forward to the final product.

**Comment by ElementalCrisis** [comment]
> On the latest downloads page the beta download link fails as the file has been deleted.

Thanks, I forgot all the old downloads were removed due to the Google incident.  

> On the installing JMM Desktop and JMM Server pages, at the bottom of the page it says use the menu on the right. It should say on the left.

Done. 

> Also I feel that any links to different sites should open in a new tab. Currently it's a mix of opening in the same tab or opening in a new tab. An example would be on the home page "What is JMM" on the "How does JMM work? section, the links to AniDB and TheTvDB both open in the same tab but the TMDb link opens in a new tab.

Agreed, I tend to make all outgoing links open in a new tab so these are ones I probably forgot. If you find any just list where and I'll make the changes.

**Comment by xsystematic** [comment]
Here's all I found (Including the ones I posted above.) I chose not to include any that link to github for 2 reasons, first I'm not sure how you would want these links to open on the final site and secondly, I'm not keen on creating extra work for you on top of everything else that needs to be finished.

index.html "Information from various sites such as AniDB, TheTvDB"
index.html. "My Anime 3 is a plugin designed for MediaPortal"
setup/myanime3.html " how to install plugins"
setup/mysql.html " MySQL Community Server"
setup/sqlserver.html "free to download."
downloads.html "Download .NET Framework 4.5 from Microsoft."
downloads.html "MyAnime3 - Installer"
downloads.html "AnimeBuddy 1.0.0.3 - Download (Windows Store link)"
support.html "My Anime 3 There's a dedicated support thread"
support.html "Plex Plugin There's a dedicated support thread"

**Comment by ElementalCrisis** [comment]
Thanks, I'll get those changed. Basically all internal links open normally and all external links open a new tab, that's how I have it setup.

**Comment by ElementalCrisis** [comment]
The new site has been launched! 

Thank you to everyone who submitted feedback and helped fixed any bugs. With the launch this thread will now be closed. 

Future development on the site can be found at https://github.com/ElementalCrisis/japanesemediamanager.github.io. Additionally, you can join us on Discord where you can get support, learn about active development of JMM and just get to know fellow users. 

https://discordapp.com/invite/0XKJW7TObKLajoKc
